### ISSUE TITLE
All processes have seccomp filters attached to them, breaking CRIU

### ISSUE THREAD

**Issue by bepvte** [microsoft/WSL#9993]
### Windows Version

Microsoft Windows [Version 10.0.22621.1555]    

### WSL Version

WSL version: 1.2.0.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.15.90.1-microsoft-standard-WSL2

### Distro Version

Debian 12

### Other Software

#9548 was reintroduced recently by some change in `init` and is causing some software like CRIU to fail.
Having seccomp filters be on all processes is not typical init behavior.
[CRIU](https://criu.org/) fails when it has seccomp filters attached to it, and these seccomp filters seem to be inherited by all the child processes of init.
This is because for a process to suspend seccomp and ptrace another process, it needs to not have seccomp filters on itself

### Repro Steps

```
$ grep Seccomp /proc/1/status
Seccomp:        2
Seccomp_filters:        1
$ sudo apt-get install -y criu
$ sudo criu check -v2 --extra
[...]
(00.047210) Error (criu/cr-check.c:750): couldn't suspend seccomp: Operation not permitted
(00.048433) Error (criu/cr-check.c:793): Dumping seccomp filters not supported: Permission denied
(00.048885) Warn  (criu/cr-check.c:813): Dirty tracking is OFF. Memory snapshot will not work.
[...]
```

### Expected Behavior

CRIU and other software that doesn't work with seccomp filters working correctly

### Actual Behavior

CRIU errors out with permission denied

**Comment by bepvte** [comment]
When updating to the prerelease 1.2.4.0 this issue no longer occurs. If that means this issue can be closed please feel free to.

**Comment by pmartincic** [comment]
Yup, this was fixed in 1.2.1

**Comment by pmartincic** [comment]
Though as fair warning, seccomp filters may exist in a potential future.
